src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,60 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 110;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (user == null) {
+            throw new RegistrationException("User can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password can't be less than 6 characters");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login can't be less than 6 characters");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null");
+        }
+        if (user.getAge() < MIN_AGE || user.getAge() > MAX_AGE) {
+            throw new RegistrationException("Not valid age: " + user.getAge()
+                    + ". Min allowed age is " + MIN_AGE
+                    + ". Max allowed age is " + MAX_AGE);
+        }
```

Comment:
Pay attention to the method `String.format()`
```suggestion
        if (user.getAge() == null || user.getAge() < MIN_AGE || user.getAge() > MAX_AGE) {
            throw new RegistrationException(String.format("Age is invalid. Min allowed age is: %s. Max allowed age is: %s", MIN_AGE, MAX_AGE);
        }
```